drivers/soundwire/bus.c

```diff
@@ -846,12 +847,18 @@ static int sdw_slave_clk_stop_callback(struct sdw_slave *slave,
 				       enum sdw_clk_stop_mode mode,
 				       enum sdw_clk_stop_type type)
 {
-	int ret;
+	struct device *dev = &slave->dev;
+	struct sdw_driver *drv;
 
-	if (slave->ops && slave->ops->clk_stop) {
-		ret = slave->ops->clk_stop(slave, mode, type);
-		if (ret < 0)
-			return ret;
+	/*
+	 * this function can only be called from a pm_runtime
+	 * sequence where the device is already locked
+	 */
+
+	if (dev->driver) {
+		drv = drv_to_sdw_driver(dev->driver);
+		if (drv && drv->ops && drv->ops->clk_stop)
```

Comment:
`drv` can not be NULL, the check can be dropped?

Reply:
yes indeed, thanks for the suggestion @ujfalusi

Reply:
done